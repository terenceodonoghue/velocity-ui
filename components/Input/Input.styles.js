import styled from '@emotion/styled';

export const Checkbox = styled.input`
  height: 0;
  width: 0;
  opacity: 0;
`;

export const Label = styled.label`
  display: inline-block;
  color: #b0bac9;
  font-size: 12px;
  font-weight: var(--font-weight-medium);
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;

export const Slider = styled.input`
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: ${props => props.width || '210px'};
  background-color: rgba(46, 91, 255, 0.15);
  outline: none;
  transition: var(--transition);

  &::-moz-range-progress {
    background-color: var(--color-clear-blue);
  }

  &::-moz-range-thumb {
    border: solid 4px var(--color-clear-blue);
    border-radius: 50%;
    height: 8px;
    width: 8px;
    background: var(--color-white);
    cursor: pointer;
  }

  &::-webkit-slider-thumb {
    appearance: none;
    border: solid 4px var(--color-clear-blue);
    border-radius: 50%;
    height: 16px;
    width: 16px;
    background: var(--color-white);
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.3;
  }
`;


export const Switch = styled.label`
  position: relative;
  display: inline-block;
  height: 20px;
  width: 36px;
  transition: var(--transition);
`;

export const Text = styled.input`
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224, 231, 255, 0.2);
  color: var(--color-dark);
  font-family: inherit;
  font-size: var(--base-font-size);
  transition: var(--transition);

  &:disabled {
    opacity: 0.3;
  }

  &:focus {
    border: solid 1px var(--color-clear-blue);
    outline: 0;
  }
`;

export const Toggle = styled.span`
  position: absolute;
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  border-radius: 10px;
  width: 36px;
  background-color: rgba(135, 152, 173, 0.4);
  cursor: pointer;
  transition: var(--transition);

  ${Checkbox}:disabled + & {
    opacity: 0.4;
  }

  ${Checkbox}:checked:enabled + & {
    background-color: var(--color-medium-green);
  }

  &::before {
    position: absolute;
    content: '';
    left: 2px;
    top: 2px;
    border-radius: 50%;
    height: 16px;
    width: 16px;
    background-color: var(--color-white);
    transition: var(--transition);

    ${Checkbox}:checked + & {
      transform: translateX(16px);
    }
  }
`;
