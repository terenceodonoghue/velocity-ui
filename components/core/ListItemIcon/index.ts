export { default } from './ListItemIcon';
export type { ListItemIconProps } from './ListItemIcon.types';
