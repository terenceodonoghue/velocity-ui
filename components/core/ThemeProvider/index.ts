export { default, defaultTheme } from './ThemeProvider';
export type { Theme } from './ThemeProvider.types';
