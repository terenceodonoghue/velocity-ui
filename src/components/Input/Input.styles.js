import styled from '@emotion/styled';
import Global from '../Global/Global';

const {
  variables: { colors, effects, typography },
} = Global;

export const CheckboxInput = styled.input`
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
`;

export const CheckboxLabel = styled.label`
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-width: 36px;
  user-select: none;
`;

export const CheckboxSwitch = styled.span`
  position: relative;
  border-radius: 10px;
  height: 20px;
  width: 36px;
  background-color: rgba(135, 152, 173, 0.4);
  cursor: pointer;
  transition: ${effects.transition};

  ${/* sc-selector */ CheckboxInput}:disabled + & {
    opacity: 0.4;
  }

  ${/* sc-selector */ CheckboxInput}:checked:enabled + & {
    background-color: ${colors.mediumGreen};
  }

  &::before {
    position: absolute;
    content: '';
    left: 2px;
    top: 2px;
    border-radius: 50%;
    height: 16px;
    width: 16px;
    background-color: ${colors.white};
    transition: ${effects.transition};

    ${/* sc-selector */ CheckboxInput}:checked + & {
      transform: translateX(16px);
    }
  }
`;

export const CheckboxText = styled.div`
  margin-right: 58px;
  font-weight: ${typography.fontWeightMedium};
  line-height: 1.47;

  p {
    margin: 0;

    &:nth-of-type(2) {
      color: ${colors.blueyGrey};
      font-weight: ${typography.fontWeightRegular};
    }
  }
`;

export const RadioInput = styled.input`
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
`;

export const RadioLabel = styled.label`
  display: flex;
  align-items: center;
  font-weight: ${typography.fontWeightMedium};
`;

export const RadioSwitch = styled.span`
  display: inline-block;
  border: solid 1px rgba(128, 151, 177, 0.5);
  border-radius: 50%;
  height: 16px;
  margin: 0 8px 0 0;
  width: 16px;
  background-color: ${colors.white};

  ${/* sc-selector */ RadioInput}:disabled + & {
    opacity: 0.4;
  }

  ${/* sc-selector */ RadioInput}:checked:enabled + & {
    border: solid 4px ${colors.clearBlue};
    background-color: ${colors.white};
  }
`;

export const SliderInput = styled.input`
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 210px;
  background-color: rgba(46, 91, 255, 0.15);
  outline: none;
  transition: ${effects.transition};

  &::-moz-range-progress {
    background-color: ${colors.clearBlue};
  }

  &::-moz-range-thumb {
    border: solid 4px ${colors.clearBlue};
    border-radius: 50%;
    height: 8px;
    width: 8px;
    background: ${colors.white};
    cursor: pointer;
  }

  &::-webkit-slider-thumb {
    appearance: none;
    border: solid 4px ${colors.clearBlue};
    border-radius: 50%;
    height: 16px;
    width: 16px;
    background: ${colors.white};
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.3;
  }
`;

export const TextInput = styled.input`
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224, 231, 255, 0.2);
  color: ${colors.dark};
  font-family: inherit;
  font-size: ${typography.baseFontSize};
  transition: ${effects.transition};

  &:disabled {
    opacity: 0.3;
  }

  &:focus {
    border: solid 1px ${colors.clearBlue};
    outline: 0;
  }
`;

export const TextLabel = styled.label`
  display: inline-block;
  color: #b0bac9;
  font-size: 12px;
  font-weight: ${typography.fontWeightMedium};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
