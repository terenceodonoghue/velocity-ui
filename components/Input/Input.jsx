/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as S from './Input.styles';

const Label = props => <S.Label {...props} />;
const Text = props => <S.Text {...props} />;
const Switch = props => (
  <S.Switch>
    <S.Checkbox type="checkbox" {...props} />
    <S.Toggle />
  </S.Switch>
);
const Slider = props => <S.Slider type="range" {...props} />;

export default {
  Label,
  Text,
  Switch,
  Slider,
};
