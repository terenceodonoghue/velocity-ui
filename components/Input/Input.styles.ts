import { css, InterpolationWithTheme } from '@emotion/core';
import { Theme } from '~/types';

export const checkboxLabel = css`
  display: flex;
  align-items: center;
  justify-content: space-between;
  min-width: 36px;
  user-select: none;

  input {
    height: 0;
    margin: 0;
    width: 0;
    opacity: 0;
  }
`;

export const checkboxSwitch: InterpolationWithTheme<Theme> = (theme) => css`
  position: relative;
  border-radius: 10px;
  height: 20px;
  width: 36px;
  background-color: rgba(135, 152, 173, 0.4);
  cursor: pointer;
  transition: ${theme.effects.transition};

  input:disabled + & {
    opacity: 0.4;
  }

  input:checked:enabled + & {
    background-color: ${theme.colors.green};
  }

  &::before {
    position: absolute;
    content: '';
    left: 2px;
    top: 2px;
    border-radius: 50%;
    height: 16px;
    width: 16px;
    background-color: ${theme.colors.white};
    transition: ${theme.effects.transition};

    input:checked + & {
      transform: translateX(16px);
    }
  }
`;

export const checkboxText: InterpolationWithTheme<Theme> = (theme) => css`
  margin-right: 58px;
  font-weight: ${theme.typography.fontWeightMedium};
  line-height: 1.47;

  p {
    margin: 0;

    &:nth-of-type(2) {
      color: ${theme.colors.blueyGrey};
      font-weight: ${theme.typography.fontWeightRegular};
    }
  }
`;

export const radioLabel: InterpolationWithTheme<Theme> = (theme) => css`
  display: flex;
  align-items: center;
  font-weight: ${theme.typography.fontWeightMedium};
  cursor: pointer;

  input {
    height: 0;
    margin: 0;
    width: 0;
    opacity: 0;
  }
`;

export const radioSwitch: InterpolationWithTheme<Theme> = (theme) => css`
  display: inline-block;
  border: solid 1px rgba(128, 151, 177, 0.5);
  border-radius: 50%;
  height: 16px;
  margin: 0 8px 0 0;
  width: 16px;
  background-color: ${theme.colors.white};

  input:disabled + & {
    opacity: 0.4;
  }

  input:checked:enabled + & {
    border: solid 4px ${theme.colors.blue};
    background-color: ${theme.colors.white};
  }
`;

export const sliderInput: InterpolationWithTheme<Theme> = (theme) => css`
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 210px;
  background-color: rgba(46, 91, 255, 0.15);
  outline: none;
  transition: ${theme.effects.transition};

  &::-moz-range-progress {
    background-color: ${theme.colors.blue};
  }

  &::-moz-range-thumb {
    border: solid 4px ${theme.colors.blue};
    border-radius: 50%;
    height: 8px;
    width: 8px;
    background: ${theme.colors.white};
    cursor: pointer;
  }

  &::-webkit-slider-thumb {
    appearance: none;
    border: solid 4px ${theme.colors.blue};
    border-radius: 50%;
    height: 16px;
    width: 16px;
    background: ${theme.colors.white};
    cursor: pointer;
  }

  &:disabled {
    opacity: 0.3;
  }
`;

export const sliderLabel = css`
  display: inline-block;
`;

export const sliderText: InterpolationWithTheme<Theme> = (theme) => css`
  display: flex;

  p {
    line-height: 1.467em;
    margin: 0;
  }

  p:nth-of-type(1) {
    flex: 1;
  }

  p:nth-of-type(2) {
    color: ${theme.colors.blueyGrey};
  }
`;

export const textInput: InterpolationWithTheme<Theme> = (theme) => css`
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224, 231, 255, 0.2);
  color: ${theme.colors.black};
  font-family: inherit;
  font-size: ${theme.typography.baseFontSize};
  transition: ${theme.effects.transition};

  &:disabled {
    opacity: 0.3;
  }

  &:focus {
    border: solid 1px ${theme.colors.blue};
    outline: 0;
  }
`;

export const textLabel: InterpolationWithTheme<Theme> = (theme) => css`
  display: inline-block;
  color: #b0bac9;
  font-size: 12px;
  font-weight: ${theme.typography.fontWeightMedium};
  letter-spacing: 1.2px;
  text-transform: uppercase;
`;
