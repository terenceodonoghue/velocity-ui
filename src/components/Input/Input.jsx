/** @jsx jsx */
import { jsx } from '@emotion/core';
import { string } from 'prop-types';
import * as S from './Input.styles';

const Radio = ({ className, label, ...rest }) => (
  <S.RadioLabel>
    <S.RadioInput type="radio" {...rest} />
    <S.RadioSwitch />
    {label}
  </S.RadioLabel>
);

const Slider = (props) => <S.SliderInput type="range" {...props} />;

const Switch = ({
  className, description, label, ...rest
}) => (
  <S.CheckboxLabel className={className}>
    {(label || description)
      && (
      <S.CheckboxText>
        {label && <p>{label}</p>}
        {description && <p>{description}</p>}
      </S.CheckboxText>
      )}
    <S.CheckboxInput type="checkbox" {...rest} />
    <S.CheckboxSwitch />
  </S.CheckboxLabel>
);

const Text = ({ className, label, ...rest }) => (
  <S.TextLabel className={className}>
    {label}
    <S.TextInput {...rest} />
  </S.TextLabel>
);

Radio.propTypes = { className: string, label: string };
Radio.defaultProps = { className: null, label: null };
Switch.propTypes = { className: string, description: string, label: string };
Switch.defaultProps = { className: null, label: null, description: null };
Text.propTypes = { className: string, label: string };
Text.defaultProps = { className: null, label: null };

export default {
  Radio,
  Slider,
  Switch,
  Text,
};
