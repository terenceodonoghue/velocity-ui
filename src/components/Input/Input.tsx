/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent, InputHTMLAttributes } from 'react';
import * as css from './Input.styles';

interface Label {
  primary: string;
  secondary: string;
}

interface InputProps extends InputHTMLAttributes<HTMLLabelElement> {
  inputProps?: InputHTMLAttributes<HTMLInputElement>;
}

export const Radio: FunctionComponent<InputProps> = ({
  children,
  inputProps,
  name,
  ...props
}) => (
  <label css={css.radioLabel} {...props}>
    <input css={css.radioInput} name={name} type="radio" {...inputProps} />
    <span css={css.radioSwitch} />
    {children}
  </label>
);

export const Slider: FunctionComponent<Label &
  InputHTMLAttributes<HTMLInputElement>> = ({
  primary,
  secondary,
  ...props
}) => (
  <label css={css.sliderLabel}>
    {(primary || secondary) && (
      <div css={css.sliderText}>
        {primary && <p>{primary}</p>}
        {secondary && <p>{secondary}</p>}
      </div>
    )}
    <input css={css.sliderInput} type="range" {...props} />
  </label>
);

export const Switch: FunctionComponent<InputProps & Label> = ({
  inputProps,
  primary,
  secondary,
  ...props
}) => (
  <label css={css.checkboxLabel} {...props}>
    {(primary || secondary) && (
      <div css={css.checkboxText}>
        {primary && <p>{primary}</p>}
        {secondary && <p>{secondary}</p>}
      </div>
    )}
    <input css={css.checkboxInput} type="checkbox" {...inputProps} />
    <span css={css.checkboxSwitch} />
  </label>
);

export const Text: FunctionComponent<InputProps> = ({
  children,
  inputProps,
  ...props
}) => (
  <label css={css.textLabel} {...props}>
    {children}
    <input css={css.textInput} {...inputProps} />
  </label>
);
