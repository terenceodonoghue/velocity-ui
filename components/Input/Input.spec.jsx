/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as Input from './Input';

describe('Input', () => {
  describe('Radio', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(withTheme(<Input.Radio />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Slider', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(withTheme(<Input.Slider min={0} max={100} />))
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Switch', () => {
    it('matches snapshot', () => {
      const tree = renderer.create(withTheme(<Input.Switch />)).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Text', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(
          withTheme(
            <Input.Text
              label="Label Text"
              placeholder="Placeholder"
              type="text"
            />,
          ),
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
