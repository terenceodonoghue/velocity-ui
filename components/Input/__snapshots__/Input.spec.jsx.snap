// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Radio matches snapshot 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 500;
  cursor: pointer;
}

.emotion-0 {
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
}

.emotion-1 {
  display: inline-block;
  border: solid 1px rgba(128,151,177,0.5);
  border-radius: 50%;
  height: 16px;
  margin: 0 8px 0 0;
  width: 16px;
  background-color: #fff;
}

.css-e5bvf:disabled + .emotion-1 {
  opacity: 0.4;
}

.css-e5bvf:checked:enabled + .emotion-1 {
  border: solid 4px #2e5bff;
  background-color: #fff;
}

<label
  className="emotion-2"
>
  <input
    className="emotion-0"
    type="radio"
  />
  <span
    className="emotion-1"
  />
</label>
`;

exports[`Input Slider matches snapshot 1`] = `
.emotion-1 {
  display: inline-block;
}

.emotion-0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border-radius: 7px;
  height: 4px;
  width: 210px;
  background-color: rgba(46,91,255,0.15);
  outline: none;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-0::-moz-range-progress {
  background-color: #2e5bff;
}

.emotion-0::-moz-range-thumb {
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 8px;
  width: 8px;
  background: #fff;
  cursor: pointer;
}

.emotion-0::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: solid 4px #2e5bff;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  background: #fff;
  cursor: pointer;
}

.emotion-0:disabled {
  opacity: 0.3;
}

<label
  className="emotion-1"
>
  <input
    className="emotion-0"
    max={100}
    min={0}
    type="range"
  />
</label>
`;

exports[`Input Switch matches snapshot 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-width: 36px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0 {
  height: 0;
  margin: 0;
  width: 0;
  opacity: 0;
}

.emotion-1 {
  position: relative;
  border-radius: 10px;
  height: 20px;
  width: 36px;
  background-color: rgba(135,152,173,0.4);
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.css-e5bvf:disabled + .emotion-1 {
  opacity: 0.4;
}

.css-e5bvf:checked:enabled + .emotion-1 {
  background-color: #33ac2e;
}

.emotion-1::before {
  position: absolute;
  content: '';
  left: 2px;
  top: 2px;
  border-radius: 50%;
  height: 16px;
  width: 16px;
  background-color: #fff;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.css-e5bvf:checked + .emotion-1::before {
  -webkit-transform: translateX(16px);
  -ms-transform: translateX(16px);
  transform: translateX(16px);
}

<label
  className="emotion-2"
>
  <input
    className="emotion-0"
    type="checkbox"
  />
  <span
    className="emotion-1"
  />
</label>
`;

exports[`Input Text matches snapshot 1`] = `
.emotion-1 {
  display: inline-block;
  color: #b0bac9;
  font-size: 12px;
  font-weight: 500;
  -webkit-letter-spacing: 1.2px;
  -moz-letter-spacing: 1.2px;
  -ms-letter-spacing: 1.2px;
  letter-spacing: 1.2px;
  text-transform: uppercase;
}

.emotion-0 {
  display: block;
  border: solid 1px #e0e7ff;
  border-radius: 5px;
  height: 38px;
  margin: 7px 0;
  max-width: 270px;
  padding: 10px 16px;
  width: 100%;
  background-color: rgba(224,231,255,0.2);
  color: #2e384d;
  font-family: inherit;
  font-size: 15px;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-0:disabled {
  opacity: 0.3;
}

.emotion-0:focus {
  border: solid 1px #2e5bff;
  outline: 0;
}

<label
  className="emotion-1"
  label="Label Text"
  placeholder="Placeholder"
  type="text"
>
  <input
    className="emotion-0"
  />
</label>
`;
